packages/flutter/test/rendering/mock_canvas.dart

```diff
@@ -1058,26 +1056,26 @@ class _RRectPaintPredicate extends _DrawCommandPaintPredicate {
     style: style,
   );
 
-  final RRect rrect;
+  final RRect? rrect;
 
   @override
   void verifyArguments(List<dynamic> arguments) {
     super.verifyArguments(arguments);
     const double eps = .0001;
     final RRect actual = arguments[0] as RRect;
     if (rrect != null &&
-       ((actual.left - rrect.left).abs() > eps ||
-        (actual.right - rrect.right).abs() > eps ||
-        (actual.top - rrect.top).abs() > eps ||
-        (actual.bottom - rrect.bottom).abs() > eps ||
-        (actual.blRadiusX - rrect.blRadiusX).abs() > eps ||
-        (actual.blRadiusY - rrect.blRadiusY).abs() > eps ||
-        (actual.brRadiusX - rrect.brRadiusX).abs() > eps ||
-        (actual.brRadiusY - rrect.brRadiusY).abs() > eps ||
-        (actual.tlRadiusX - rrect.tlRadiusX).abs() > eps ||
-        (actual.tlRadiusY - rrect.tlRadiusY).abs() > eps ||
-        (actual.trRadiusX - rrect.trRadiusX).abs() > eps ||
-        (actual.trRadiusY - rrect.trRadiusY).abs() > eps)) {
+       ((actual.left - rrect!.left).abs() > eps ||
+        (actual.right - rrect!.right).abs() > eps ||
+        (actual.top - rrect!.top).abs() > eps ||
+        (actual.bottom - rrect!.bottom).abs() > eps ||
+        (actual.blRadiusX - rrect!.blRadiusX).abs() > eps ||
+        (actual.blRadiusY - rrect!.blRadiusY).abs() > eps ||
+        (actual.brRadiusX - rrect!.brRadiusX).abs() > eps ||
+        (actual.brRadiusY - rrect!.brRadiusY).abs() > eps ||
+        (actual.tlRadiusX - rrect!.tlRadiusX).abs() > eps ||
+        (actual.tlRadiusY - rrect!.tlRadiusY).abs() > eps ||
+        (actual.trRadiusX - rrect!.trRadiusX).abs() > eps ||
+        (actual.trRadiusY - rrect!.trRadiusY).abs() > eps)) {
```

Comment:
It's probably not critical and maybe the compiler should just figure this out anyway, but for my curiosity - would it be better to just have a single copy of `rrect` that's asserted not null here at the top? like 

```dart
final RRect rrect = this.rrect!;
```

or something to that effect?

Reply:
I don't know. I think doing it the other way would in this case actually make the code more complicated because you would have two separate ifs.

Reply:
Yeah. And honestly if the compiler doesn't optimize this the way we'd want I'd be pretty disappointed.